File: 1790. Check if One String Swap Can Make Strings Equal/solution.java
class Solution {
  public boolean areAlmostEqual(String s1, String s2) {
    List<Integer> different = new ArrayList<Integer>();
    for (int index = 0; index < s1.length(); index++){
      if (s1.charAt(index) != s2.charAt(index)){
        different.add(index);
      }
      if (different.size() > 2){
        return false;
      }
    }
    return different.size() == 0 || (different.size() == 2 && s1.charAt(different.get(0)) == s2.charAt(different.get(1)) && s1.charAt(different.get(1)) == s2.charAt(different.get(0)));
  }
}

File: 1790. Check if One String Swap Can Make Strings Equal/solution.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
 var areAlmostEqual = function(s1, s2) {
  let different = [];
  for (let index = 0; index < s1.length; index++){
    if (s1[index] != s2[index]){
      different.push(index);
    }
    if (different.length > 2){
      return false;
    }
  }
  return different.length == 0 || (different.length == 2 && s1[different[0]] == s2[different[1]] && s1[different[1]] == s2[different[0]]);
};